### ISSUE TITLE
Add rule to disallow `cursor` for `:hover`

### ISSUE THREAD

**Issue by Maxim-Mazurok** [stylelint/stylelint#6739]
### What is the problem you're trying to solve?

I found this code in CSS to make a pointy hand cursor for an element:

```css
.my-class:hover {
  cursor: pointer
}
```

I want to always prefer this:
```css
.my-class {
  cursor: pointer
}
```

Reasons:
1. It does essentially the same thing in less code. There's no point in adding `:hover` for setting `cursor` behavior in 99.99% of cases. Because only `cursor` value from the currently hovered element is ever applied.
2. It makes it harder to debug code. I spent like 15 minutes trying to figure out where we're setting cursor value because in Chrome dev tool inspector you can't see `:hover` values unless you force hover state:
	![image](https://user-images.githubusercontent.com/7756211/228096429-0551f81c-81eb-427e-ad19-9947a9e6539f.png)

### What solution would you like to see?

A rule that would at a minimum prevent usage of `cursor` inside of `*:hover`
That probably can be later extended to `pointer-events`, `:focus`, and others.

**Comment by ybiquitous** [comment]
@Maxim-Mazurok Thanks for the interesting proposal. The proposed rule should be categorized in [Enforce conventions](https://stylelint.io/user-guide/rules/#enforce-conventions).

Considering [our criteria for a rule](https://stylelint.io/developer-guide/rules#define-the-rule), I don't know whether disallowing the `cursor` property inside a `*:hover` selector is a common pattern. 🤔 

For example, it might be more helpful to add a more general rule that disallows the use of specific properties inside specific selectors, instead of disallowing `cursor`.

So, I'd like to hear several opinions.

**Comment by Maxim-Mazurok** [comment]
> I don't know whether disallowing the `cursor` property inside a `*:hover` selector is a common pattern. 🤔

It might not be common, but I don't think it matters. This is a pretty much "lossless" optimization. The `cursor` property is only applied when the element is being `:hover`ed. The only case when I think it might be inconvenient is when you have something like this:
```css
.test {} /* no rules, can be removed */
.test:hover {
  color: red;
  cursor: hover;
}
```
As you can see in that case devs might prefer to put `cursor` in `:hover` instead of adding `.test {}` selector just for `cursor`. But as described in the OP, this will result in layout that is harder to debug, so I'd argue it should be avoided.

> So, I'd like to hear several opinions.

My co-workers seem to agree and upvoted the proposal. Who else shall we tag?

**Comment by ybiquitous** [comment]
I understand your motivation, but I feel it is hard to add a new rule for this specific case. Reasons:

- it is possible to realize this case by the existing rule [`rule-selector-property-disallowed-list`](https://stylelint.io/user-guide/rules/rule-selector-property-disallowed-list/)
- there is no rule for a specific property (e.g. `cursor`) in the [built-in rules](https://stylelint.io/user-guide/rules/) (we provide a [plugin system](https://stylelint.io/developer-guide/plugins) for such a purpose)

```json
{
  "rules": {
    "rule-selector-property-disallowed-list": {
      "/:(hover|focus)/": ["cursor"]
    }
  }
}
```

<img width="744" alt="Screenshot 2023-04-09 at 22 29 35" src="https://user-images.githubusercontent.com/473530/230775584-523e697e-3db6-4cb2-b347-b8e38b41acc1.png">


See the [demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIB0BbAngWjAGwIYGd94ALCANxgCcACYAHQDtrqwBXS-CS+agBwgCWDAC5UA3IwC+jRqkw4CRAGYQ2+Wo2ZsOXHvyGjKEhtIYgANCCUDsMAHK4UcRDAAej3raRhCF8BAZrAHMEEHomajoQSlZbfCiecOZmKJjbDHwYWzBhLgxeSgheKmFMKAF8XGxsCAB3GCgMbArhBI0I5MiQAHp4AApSCkoAHxU1AEputoBtKO1OSiiAXU1k02ZTST9IQIEggDEuFFxhUIArTjNLWF58UKSu-FLbZpE2qLxRJ6jzVaintAvAxYALBDLCXAMKC4ShQd4gT4wb4gKQgSRAA).

**Comment by Maxim-Mazurok** [comment]
Ah, I see, thanks for that! You might want to consider adding this to the standard config as I think most people would benefit from it. Otherwise yeah, I would probably agree that no need for a separate rule... Still having a separate one might make it easier to find and improve DX, but I guess this isn't critical.

**Comment by ybiquitous** [comment]
> You might want to consider adding this to the standard config as I think most people would benefit from it.

That might also be one idea. But, if adding it to the standard config, the message "Unexpected property "cursor" ..." sounds a bit not kind to me. 🤔 

If we also consider customizing a message, adding it to the standard config might be too much.

But I will keep this issue open to hear people's opinions.

**Comment by Mouvedia** [comment]
I agree with @ybiquitous, this is too specific to warrant a new rule.

**Comment by Maxim-Mazurok** [comment]
Fair enough, we'll use https://github.com/stylelint/stylelint/issues/6739#issuecomment-1501129665 suggestion, unsubscribing for now, but feel free to keep it open or close, cheers!

**Comment by ybiquitous** [comment]
@Maxim-Mazurok Thanks for the response. I'm closing this issue.

If you create a plugin, please consider adding it to [Awesome Stylelint](https://github.com/stylelint/awesome-stylelint)!
